cmd/cosign/cli/verify/verify_attestation.go

```diff
@@ -112,14 +111,19 @@ func (c *VerifyAttestationCommand) Exec(ctx context.Context, images []string) (e
 		if err != nil {
 			return fmt.Errorf("error reading certification chain path file: %w", err)
 		}
-		// TODO: Update this logic once https://github.com/sigstore/timestamp-authority/issues/121 gets merged.
-		// This relies on untrusted leaf certificate.
-		tsaCertPool := x509.NewCertPool()
-		ok := tsaCertPool.AppendCertsFromPEM(pemBytes)
-		if !ok {
-			return fmt.Errorf("error parsing response into Timestamp while appending certs from PEM")
+
+		leaves, intermediates, roots, err := tsa.SplitPEMCertificateChain(pemBytes)
```

Comment:
Not necessary to address in this pull request but it looks like this `Exec` function and the `Exec` function in `cmd/cosign/cli/verify/verify.go` have a lot of repeating logic?

Reply:
+1 I'll create a function and call it from the different places.

Reply:
Yea, all are looking similar, I think this was brought up in another PR recently. Let's do that as a follow up.